modules/shortcodes/vimeo.php

```diff
@@ -172,6 +172,77 @@ function vimeo_shortcode( $atts ) {
 
 add_shortcode( 'vimeo', 'vimeo_shortcode' );
 
+/**
+ * Filters the Vimeo shortcode to be AMP-compatible.
+ *
+ * @param string $html The shortcode HTML.
+ * @param string $shortcode_tag The shortcode's tag (name).
+ * @param array  $attr The attributes of the shortcode.
+ * @return string The filtered HTML.
+ */
+function amp_vimeo_shortcode( $html, $shortcode_tag, $attr ) {
+	if (
+		( class_exists( 'Jetpack_AMP_Support' ) && ! Jetpack_AMP_Support::is_amp_request() )
+		||
+		'vimeo' !== $shortcode_tag
+	) {
+		return $html;
+	}
+
+	$video_id = null;
+	if ( isset( $attr[0] ) ) {
+		$video_id = jetpack_shortcode_get_vimeo_id( $attr );
+	} elseif ( isset( $attr['id'] ) ) {
+		$video_id = $attr['id'];
+	}
```

Comment:
Should we bail if we don't have a video ID by that point, or if it is null or `0`, instead of waiting until later?

Reply:
Good idea, applied in https://github.com/Automattic/jetpack/pull/13921/commits/cb09ba40f08edf736c752d7bc6cc06be1a30e5e4#diff-35236fb9fce8df90f9eea45af69cc92cR199